pages/_app.tsx

```diff
@@ -1,20 +1,39 @@
-import { UserProvider } from '@auth0/nextjs-auth0';
-import { AppProps } from 'next/app';
+import { UserProfile, UserProvider } from '@auth0/nextjs-auth0';
+import { NextComponentType } from 'next';
+import { AppContext, AppProps } from 'next/app';
 import Head from 'next/head';
-
-import { SidebarProvider } from '~/lib/context/sidebar_context';
-import { FormProvider } from '~/lib/context/form_context';
-
-import { FC } from 'react';
 import { SWRConfig } from 'swr';
 import Layout from '~/components/Layout';
+import { FormProvider } from '~/lib/context/form_context';
+import { SidebarProvider } from '~/lib/context/sidebar_context';
 import { fetcher } from '~/lib/swr';
+import { Site, SiteList } from '~/lib/types';
 import '~/styles/globals.css';
 
-const App: FC<AppProps> = ({ Component, pageProps }) => {
+type InitialProps = { siteList?: SiteList };
+
+type SitesAppType = NextComponentType<
+  AppContext,
+  InitialProps,
+  AppProps<{ user: UserProfile }> & InitialProps
+>;
+
+const SitesApp: SitesAppType = ({ Component, pageProps, siteList }) => {
```

Comment:
I'm thinking that `SitesApp` might not be the best name for this—why not just keep it as `App`? It does retrieve sites on server side, but it also still controls everything else.